source/uwp/SharedRenderer/lib/Util.cpp

```diff
@@ -535,19 +536,50 @@ bool IsBackgroundImageValid(winrt::AdaptiveBackgroundImage const& backgroundImag
     return false;
 }
 
+winrt::Uri UriTryCreate(winrt::hstring const &uri, winrt::hstring const& baseUri = L"")
+{
+    auto factory = winrt::
+        get_activation_factory<winrt::Windows::Foundation::Uri, winrt::Windows::Foundation::IUriRuntimeClassFactory>();
```

Comment:
@beervoley, are there any reasons we have to do this way instead of creating using winrt::Uri?

Reply:
in cpp winrt we don't have a constructor for `winrt::Uri` that doesn't throw if the `uri` is invalid thus every tiem we try to create `winrt::Uri` and there's something wrong with the `uri` -> it throws

Reply:
thanks @beervoley